orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/V1SchemaExecutionGenerator.java

```diff
@@ -155,4 +153,39 @@ private void addTriggers(
               .orElse(Collections.emptyList()));
     }
   }
+
+  private void addExpectedArtifacts(
+      Map<String, Object> pipeline,
+      PipelineTemplate template,
+      TemplateConfiguration configuration,
+      TemplatedPipelineRequest request) {
+
+    List<HashMap<String, Object>> mergedExpectedArtifacts =
+        mergeExpectedArtifact(
+            emptyListIfNull(configuration.getConfiguration().getExpectedArtifacts()),
+            emptyListIfNull(request.getExpectedArtifacts()));
+
+    if (configuration.getConfiguration().getInherit().contains("expectedArtifacts")) {
```

Comment:
nit: i find this a bit cleaner/easier to parse:

```
	  if (configuration.getConfiguration().getInherit().contains("expectedArtifacts")) {
      mergedExpectedArtifacts = mergeExpectedArtifact(
              emptyListIfNull(template.getConfiguration().getExpectedArtifacts()),
              mergedExpectedArtifacts));
    }

    pipeline.put("expectedArtifacts", mergedExpectedArtifacts);
```

Reply:
addressed